ts/screens/onboarding/EmailInsertScreen.tsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * A screen where user after login (with CIE) can set email address if it is
+ * not present in the profile.
+ */
+
+import { none, Option, some } from "fp-ts/lib/Option";
+import * as pot from "italia-ts-commons/lib/pot";
+import { EmailString } from "italia-ts-commons/lib/strings";
+import { untag } from "italia-ts-commons/lib/types";
+
+import { Content, Form, Text, View } from "native-base";
+
+import * as React from "react";
+
+import {
+  Alert,
+  KeyboardAvoidingView,
+  Platform,
+  StyleSheet
+} from "react-native";
+import { NavigationScreenProp, NavigationState } from "react-navigation";
+import { connect } from "react-redux";
+import { LabelledItem } from "../../components/LabelledItem";
+import BaseScreenComponent from "../../components/screens/BaseScreenComponent";
+import FooterWithButtons from "../../components/ui/FooterWithButtons";
+import H4 from "../../components/ui/H4";
+import Markdown from "../../components/ui/Markdown";
+import I18n from "../../i18n";
+import { abortOnboarding, emailInsert } from "../../store/actions/onboarding";
+import { ReduxProps } from "../../store/actions/types";
+import { GlobalState } from "../../store/reducers/types";
+import customVariables from "../../theme/variables";
+
+type OwnProps = {
+  navigation: NavigationScreenProp<NavigationState>;
+};
+
+type Props = ReduxProps & OwnProps & ReturnType<typeof mapStateToProps>;
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1
+  },
+  content: {
+    flex: 1
+  },
+  horizontalPadding: {
+    paddingHorizontal: customVariables.contentPadding
+  },
+  boldH4: {
+    fontWeight: customVariables.textBoldWeight,
+    paddingTop: customVariables.spacerLargeHeight
+  },
+  icon: {
+    marginTop: Platform.OS === "android" ? 4 : 6 // correct icon position to align it with baseline of email text}
+  },
+  emailInput: {
+    fontWeight: customVariables.h1FontWeight,
+    color: customVariables.h1Color,
+    fontSize: 18
+  }
+});
+
+const EMPTY_EMAIL = "";
+
+type State = Readonly<{
+  email: Option<string>;
+}>;
+
+/**
+ * A screen to allow user to insert an email address.
+ */
+class EmailInsertScreen extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = { email: this.props.email };
+  }
+
+  /**
+   * Footer
+   */
+  private renderFooterButtons() {
+    const continueButtonProps = {
+      disabled: this.isValidEmail() !== true,
```

Comment:
with the change proposed
```suggestion
      disabled: !this.isValidEmail(),
```

Reply:
as mentioned before, also undefined should be provided as a different state. See comment@99

Would you want me to change empty input behaviour?

Reply:
@lmorelli986 your approach is right. I didn't think about that case